File: Easy/14. Longest Common Prefix/LongestCommonPrefix.py
'''
LeetCode Easy: 14. Longest Common Prefix
https://leetcode.com/problems/longest-common-prefix/

Check LongestCommonPrefix.cpp for brief solution explanation.
'''
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        min_len = len(strs[0])
        for i in range(len(strs)):
            min_len = min(min_len, len(strs[i]))
        for i in range(min_len):
            all_same = True
            c = strs[0][i]
            for j in range(len(strs)):
                if c!=strs[j][i]:
                    all_same = False
                    break
            if not all_same:
                break
            prefix+=c
        return prefix

File: Easy/14. Longest Common Prefix/LongestCommonPrefix.java
/*
LeetCode Easy: 14. Longest Common Prefix
https://leetcode.com/problems/longest-common-prefix/

Check LongestCommonPrefix.cpp for brief solution explanation.
*/
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String prefix = "";
        for(int i = 0;;i++){
            boolean all_same = true;
            char c = ' ';
            for(int j = 0;j<strs.length;j++){
                if(strs[j].length()<=i){
                    all_same = false;
                    break;
                }else if(c==' '){
                    c = strs[j].charAt(i);
                }else if(c!=strs[j].charAt(i)){
                    all_same = false;
                    break;
                }
            }
            if(!all_same){
                break;
            }
            prefix+=c;
        }
        return prefix;
    }
}